tar/siginfo.c

```diff
@@ -191,3 +260,23 @@ siginfo_done(struct bsdtar *bsdtar)
 	/* Free internal data structure. */
 	free(bsdtar->siginfo);
 }
+
+/**
+ * siginfo_raisetotal(total):
+ * Look at how many bytes on disk have been processed since the last update,
+ * and trigger a siginfo_printinfo() if it exceeds tarsnap_opt_processbytes.
+ */
+static void
+siginfo_raisetotal(uint64_t total)
+{
+	static uint64_t lastprogress = 0;
+
+	if (tarsnap_opt_progressbytes != (uint64_t)(-1)) {
+		if (total > lastprogress + tarsnap_opt_progressbytes) {
+			lastprogress = total;
+
+			/* Fake a SIGINFO (no need for an actual signal). */
+			siginfo_received = 1;
```

Comment:
Maybe a dumb question, but is there a reason to not call `siginfo_printinfo` here?

Reply:
Oh, we don't have the `struct bsdtar` here.  But that just raises another question, why isn't this just part of `siginfo_setinfo`? 

And if we do that, `lastprogress` can be part of the siginfo structure rather than being a random static variable here...

Reply:
I used `static uint64_t lastprogress` because that's what happened in `raisesigs()` from `tar/storage/storage_write.c' (with `lastcheckpoint`).  I thought it was a nice solution, actually -- it keeps the variable scope isolated to the code that actually needs it.  Are you sure that you'd rather have it inside `struct siginfo_data`?

Good point about moving it into `siginfo_setinfo()`.